app/src/androidTest/java/com/github/steroidteam/todolist/NoteDisplayActivityTest.java

```diff
@@ -0,0 +1,84 @@
+package com.github.steroidteam.todolist;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.intent.Intents.intending;
+import static androidx.test.espresso.intent.matcher.IntentMatchers.hasAction;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+
+import android.app.Activity;
+import android.app.Instrumentation;
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.net.Uri;
+import android.os.Bundle;
+import android.os.Parcelable;
+import androidx.test.espresso.intent.rule.IntentsTestRule;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.rule.GrantPermissionRule;
+import com.github.steroidteam.todolist.view.NoteDisplayActivity;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(AndroidJUnit4.class)
+public class NoteDisplayActivityTest {
+
+    @Rule
+    public IntentsTestRule<NoteDisplayActivity> mIntentsRule =
+            new IntentsTestRule<>(NoteDisplayActivity.class);
+
+    @Rule
+    public GrantPermissionRule permissionRule =
+            GrantPermissionRule.grant(android.Manifest.permission.WRITE_EXTERNAL_STORAGE);
+
+    @Before
+    public void stubImagePickerIntent() {
+        savePickedImage();
+        Instrumentation.ActivityResult result = createImageGallerySetResultStub();
+        intending(hasAction(Intent.ACTION_CHOOSER)).respondWith(result);
+    }
+
+    @Test
+    public void filePickingWorks() {
+        onView(withId(R.id.camera_button)).perform(click());
+    }
+
+    private void savePickedImage() {
+        Bitmap bm =
```

Comment:
It's not super clear why are we using this image. I guess that the idea is to reuse something that we already have lying around the resources, but it might be better to explicitly add something with a more likely size (e.g. a 1MB picture), or to generate it procedurally as explained below.